src/ImageSharp/Formats/Webp/WebpAnimationDecoder.cs

```diff
@@ -0,0 +1,408 @@
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            AnimationFrameData frameData = this.ReadFrameHeader(stream);
+            long streamStartPosition = stream.Position;
+
+            WebpChunkType chunkType = WebpChunkParsingUtils.ReadChunkType(stream, this.buffer);
+            bool hasAlpha = false;
+            byte alphaChunkHeader = 0;
+            if (chunkType is WebpChunkType.Alpha)
+            {
+                alphaChunkHeader = this.ReadAlphaData(stream);
+                hasAlpha = true;
+                chunkType = WebpChunkParsingUtils.ReadChunkType(stream, this.buffer);
+            }
+
+            WebpImageInfo webpInfo = null;
+            var features = new WebpFeatures();
+            switch (chunkType)
+            {
+                case WebpChunkType.Vp8:
+                    webpInfo = WebpChunkParsingUtils.ReadVp8Header(this.memoryAllocator, stream, this.buffer, features);
+                    features.Alpha = hasAlpha;
+                    features.AlphaChunkHeader = alphaChunkHeader;
+                    break;
+                case WebpChunkType.Vp8L:
+                    webpInfo = WebpChunkParsingUtils.ReadVp8LHeader(this.memoryAllocator, stream, this.buffer, features);
+                    break;
+                default:
+                    WebpThrowHelper.ThrowImageFormatException("Read unexpected chunk type, should be VP8 or VP8L");
+                    break;
+            }
+
+            ImageFrame<TPixel> currentFrame = null;
+            ImageFrame<TPixel> imageFrame;
+            if (previousFrame is null)
+            {
+                image = new Image<TPixel>(this.configuration, (int)width, (int)height, backgroundColor.ToPixel<TPixel>(), this.metadata);
+
+                this.SetFrameMetadata(image.Frames.RootFrame.Metadata, frameData.Duration);
+
+                imageFrame = image.Frames.RootFrame;
+            }
+            else
+            {
+                currentFrame = image.Frames.AddFrame(previousFrame); // This clones the frame and adds it the collection.
+
+                this.SetFrameMetadata(currentFrame.Metadata, frameData.Duration);
+
+                imageFrame = currentFrame;
+            }
+
+            int frameX = (int)(frameData.X * 2);
+            int frameY = (int)(frameData.Y * 2);
+            int frameWidth = (int)frameData.Width;
+            int frameHeight = (int)frameData.Height;
+            var regionRectangle = Rectangle.FromLTRB(frameX, frameY, frameX + frameWidth, frameY + frameHeight);
+
+            if (frameData.DisposalMethod is AnimationDisposalMethod.Dispose)
+            {
+                this.RestoreToBackground(imageFrame, backgroundColor);
+            }
+
+            using Image<TPixel> decodedImage = this.DecodeImageData<TPixel>(frameData, webpInfo);
```

Comment:
Do we need to use `Image<TPixel>` here? `Buffer2D<TPixel>` would be more efficient.

Reply:
Also changed with 2a8418f9